code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingInternal.java

```diff
@@ -471,6 +502,221 @@ private static void addXDMData(final EventData eventData, final Map<String, Obje
+    /**
+     * @param event A Messaging Request Content {@link Event}.
+     * @return {@code boolean} indicating if the passed in event is a message fetch event.
+     */
+    private boolean isFetchMessagesEvent(final Event event) {
+        if (event == null || event.getEventData() == null) {
+            return false;
+        }
+
+        return event.getEventData().containsKey(MessagingConstant.EventDataKeys.Messaging.REFRESH_MESSAGES)
+                && MessagingConstant.EventType.MESSAGING.equalsIgnoreCase(event.getType())
+                && EventSource.REQUEST_CONTENT.getName().equalsIgnoreCase(event.getSource());
+    }
+
+    /**
+     * @param event A Rules Response Content {@link Event}.
+     * @return {@code boolean} indicating if the passed in event is a messaging consequence event.
+     */
+    private boolean isMessagingConsequenceEvent(final Event event) {
+        if (event == null || event.getEventData() == null) {
+            return false;
+        }
+
+        return event.getEventData().containsKey(MessagingConstant.EventDataKeys.RulesEngine.CONSEQUENCE_TRIGGERED)
+                && EventType.RULES_ENGINE.getName().equalsIgnoreCase(event.getType())
+                && EventSource.RESPONSE_CONTENT.getName().equalsIgnoreCase(event.getSource());
+    }
+
+    /**
+     * Generates and dispatches an event prompting the Personalization extension to fetch in-app messages.
+     */
+    private void fetchMessages() {
+        // create event to be handled by offers
+        final ArrayList<Object> decisionScopes = new ArrayList<>();
+        final HashMap<String, Object> offersIdentifiers = new HashMap<>();
+        offersIdentifiers.put(MessagingConstant.EventDataKeys.Offers.ITEM_COUNT, Variant.fromInteger(MAX_ITEM_COUNT));
+        offersIdentifiers.put(MessagingConstant.EventDataKeys.Offers.ACTIVITY_ID, Variant.fromString(POC_ACTIVITY_ID_MULTI));
+        offersIdentifiers.put(MessagingConstant.EventDataKeys.Offers.PLACEMENT_ID, Variant.fromString(POC_PLACEMENT_ID_MULTI));
+        decisionScopes.add(offersIdentifiers);
+
+        final HashMap<String, Object> eventData = new HashMap<>();
+        eventData.put(MessagingConstant.EventDataKeys.Offers.TYPE, MessagingConstant.EventDataKeys.Offers.PREFETCH);
+        eventData.put(MessagingConstant.EventDataKeys.Offers.DECISION_SCOPES, decisionScopes);
+
+        final Event messageFetchEvent = new Event.Builder(MessagingConstant.EventName.MESSAGING_REFRESH_IAM, MessagingConstant.EventType.OFFERS, EventSource.REQUEST_CONTENT.getName())
+                .setEventData(eventData)
+                .build();
+
+        // send event
+        MobileCore.dispatchEvent(messageFetchEvent, new ExtensionErrorCallback<ExtensionError>() {
+            @Override
+            public void error(ExtensionError extensionError) {
+                Log.error(LOG_TAG, "%s - Error in dispatching event for fetching messages from Offers.", SELF_TAG);
```

Comment:
in alignment with our logging guidelines, i think all of these error callbacks should just be logged as warnings